src/Chapter3.hs

```diff
@@ -971,6 +1069,24 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data day = Mon | Tue | Wed | Thu | Fri | Sat | Sun
+
+isWeekend :: day -> Bool
+isWeekend w = case w of
+  Sat -> True
+  Sun -> True
+  _ -> False
+
+nextDay :: day -> day
+nextDay d
+  | d == maxBound = minBound
+  | otherwise = succ d
+
+daysToParty :: day -> Int
+daysToParty day
+  | day > Friday = (fromEnum day - fromEnum Friday) + fromEnum Friday
+  | otherwise = fromEnum Friday - fromEnum day
```

Comment:
Nice solution! You can implement it even shorter, if you remember from Chapter One, how `mod` works for negative numbers :slightly_smiling_face: